Sources/SwiftGraphQLClient/Client/Selection.swift

```diff
@@ -122,16 +122,30 @@ extension OperationResult {
     fileprivate func decode<T, TypeLock>(
         selection: Selection<T, TypeLock>
     ) throws -> DecodedOperationResult<T> {
-        let data = try selection.decode(raw: self.data)
-        
-        let result = DecodedOperationResult(
-            operation: self.operation,
-            data: data,
-            errors: self.errors,
-            stale: self.stale
-        )
-        
-        return result
+        if self.data == nil && !self.errors.isEmpty {
+          throw CombinedError.combined(errors)
+        }
+
+        do {
+            let data = try selection.decode(raw: self.data)
+
+            let result = DecodedOperationResult(
+                operation: self.operation,
+                data: data,
+                errors: self.errors,
+                stale: self.stale
+            )
+
+            return result
+        } catch {
+            if self.errors.isEmpty {
+                throw error
+            } else {
+                var errors = self.errors
+                errors.append(.parsing(error))
```

Comment:
NOTE TO SELF: It makes sense to say that `.parsing` errors have precedence over all other errors (i.e. "if we encounter a parsing error it is okay to disregard all other errors arising during execution").

If you can't decode a result you most likely haven't decoded any meaningful errors anyway.